Import users ui

Pratt0923 opened the pull request.

Pratt0923 committed:
creating users, positions, and programs from xlsx data

Pratt0923 committed:
Merge branch 'master' of https://github.com/rubyforgood/doublehranch into sanitization

Pratt0923 committed:
can import users, positions, and programs fully now

Pratt0923 committed:
Merge branch 'master' of https://github.com/rubyforgood/doublehranch into sanitization

Pratt0923 committed:
Merge branch 'master' of https://github.com/rubyforgood/doublehranch into sanitization

Pratt0923 committed:
import from spreadsheet is completed

Pratt0923 committed:
removed uncessary test

Pratt0923 committed:
ui for importing users

Pratt0923 committed:
merge

Pratt0923 committed:
added routes for user importing

Pratt0923 committed:
merged changes with routes

Pratt0923 committed:
finished basic functionality of ui design for uploader

Pratt0923 committed:
ui polished a little

Pratt0923 committed:
removed space from seperator in normalize year method

zph commented on spec/lib/user_importer_spec.rb:
```diff
@@ -89,22 +89,18 @@
         it 'normalizes column names' do
-          actual = subject.normalize_column_names.first["Year"]
+          actual = subject.sanitized_rows.first["Year"]
           expect(actual).not_to eq(nil)
```
Let's assert for what the thing `is` vs that it's not `nil`

zph commented on spec/lib/user_importer_spec.rb:
```diff
@@ -89,22 +89,18 @@
-        expect(subject.import_by_row).to eq()
+      it 'imports things' do
+        subject.import_by_row
+        expect(User.all.count).to_not eq(0)
```
Similar idea here, maybe we assert that the final user matches some characteristic or that we were able to import `User.all.count == 101` users (unsure how many we have).

zph commented on spec/lib/user_importer_spec.rb:
```diff
@@ -89,22 +89,18 @@
 
       it 'normalizes column names' do
-        expect(subject.normalize_column_names.first["Year"]).not_to eq(nil)
+        expect(subject.sanitized_rows.first["Year"]).not_to eq(nil)
```
Assert to what we really expect vs that it's `not nil`

zph reviewed (changes requested):
Looking good overall. Comments on the specs and will try it out in browser.

Do we need to limit permissions to the user route? Is it already restricted to only admin users being allowed to upload the XLSX?

Pratt0923 committed:
fixing position relationships that are nil

Pratt0923 committed:
user uploads by admin polished and can edit now

zph committed:
Sanitize privacy_settings to be bool and rewrite `.not_to` specs

Positively assert what values should be and normalize privacy_settings
to the bool value associated with `1` or `0` or `true`.

zph committed:
Merge branch 'master' into import_users_ui

zph committed:
Add note about hstore behavior in User model

zph committed:
Adds feedback to user when importer has invalid columns
